### ISSUE TITLE
In TastyReflect, PackageDef#members never works

### ISSUE THREAD

**Issue by AleksanderBG** [lampepfl/dotty#9503]
All package symbols when loaded from Tasty-Reflect seem to have JavaDefined flag set. This means that we cannot list any member of any package, due to implemenetaion of `.member`.

Even if we could, package members seem to contain weirdly unexpected symbols. All classes seem to have a companion module and a companion module's class, even if source does not contain them. One could think that such symbols would have empty pos. However, it seems that many symbols which are defined in source have empty pos as well.

See:
https://github.com/lampepfl/dotty/compare/master...dotty-staging:wip/tasty-inspector/unexpected-symbols
https://github.com/lampepfl/dotty/commit/5d22e0b537480599523f02d9dc43cc8d13c935d3#diff-1eb59a7a3c5edcc15bf100f186f1a220R100-R115
https://github.com/lampepfl/dotty/commit/5d22e0b537480599523f02d9dc43cc8d13c935d3#diff-1eb59a7a3c5edcc15bf100f186f1a220R154-R162

**Comment by smarter** [comment]
> many symbols which are defined in source have empty pos as well.

Yeah, this can happen because the `coord` parameter of `newSymbol` has a default value, so it can be easy to forget to set it correctly, imo we should remove that default value and be explicit about symbol positions. Anyway, `.is(Synthetic)` should be able to distinguish between user-defined and compiler-defined symbols.

**Comment by AleksanderBG** [comment]
@smarter those symbols don't have the Synthetic flag, see https://github.com/lampepfl/dotty/commit/5d22e0b537480599523f02d9dc43cc8d13c935d3#diff-1eb59a7a3c5edcc15bf100f186f1a220R154-R162 - only class Method is present in source, but for some reason we get a module Method and a module class Method$ as well. Neither has the Synthetic flag.

**Comment by AleksanderBG** [comment]
Also, IDK if this is relevant or not, but on 0.26.0-RC1 accesing .flags of such symbols says that `.complete` is not implemented and crashes.

**Comment by nicolasstucki** [comment]
How can this be reproduced?

**Comment by abgruszecki** [comment]
@nicolasstucki a branch with a test is linked. Do you want me to rebase it onto master?
